s2fft/precompute_transforms/spherical.py

```diff
@@ -518,7 +442,9 @@ def forward_transform_jax(
 
     flm = jnp.zeros(samples.flm_shape(L), dtype=jnp.complex128)
     flm = flm.at[:, m_start_ind:].set(
-        jnp.einsum("...tlm, ...tm -> ...lm", kernel, ftm, optimize=True)
+        jnp.einsum(
+            "...tlm, ...tm -> ...lm", kernel.astype(flm.dtype), ftm, optimize=True
```

Comment:
Explicit `dtype` cast here was needed to avoid getting warnings when computing gradients through this function which were causing test failures, which indicated an implicit cast from complex to real types was happening which might be loosing information. I think this was due to the `kernel` argument here being of real type but as `ftm` is complex and the `flm` output of the `einsum` is complex, `kernel` will be implicitly cast to complex as part of `einsum` and in the reverse-pass the derivatives with respect to the `kernel` will therefore be complex even though we only should retain the real valued component.